utils/handleError.ts

```diff
@@ -23,12 +23,68 @@ export function handleGiftError(e: unknown) {
   }
 }
 
-export function handleUserError(e: unknown) {
+type KnownFrontEndErrorTexts =
+  | 'email was not unique!'
+  | 'invalid credentials!'
+  | 'invalid request body!'
+  | (string & Record<never, never>);
+
+const FRONT_END_HANDLER: Record<KnownFrontEndErrorTexts, string> = {
+  'email was not unique!': 'Sähköposti on jo käytössä',
+  'invalid credentials!': 'Sähköposti tai salasana on virheellinen!',
+  'invalid request body!': 'Sähköposti tai salasana on virheellinen',
+};
```

Comment:
Olisiko tälle parempaa toteutusta :smile: 

Lyhykäisyydessään tuon tehtävä on kääntää serveriltä tulevat virheet (ovat englanniksi) suomeksi. Onko tuon tekemiseen jokin parempi tapa kuin käyttää `Record`ia?

Jos `Record` on paras tapa toteuttaa, teoriassa lienee parempi tehdä ilman `KnownFrontEndErrorTexts`-typeä?

Tähän tapaan:
```suggestion

const FRONT_END_HANDLER: Record<string, string> = {
  'email was not unique!': 'Sähköposti on jo käytössä',
  'invalid credentials!': 'Sähköposti tai salasana on virheellinen!',
  'invalid request body!': 'Sähköposti tai salasana on virheellinen',
};
```

Reply:
Itse tekisin tuon virheen määrityksen näin

```ts
const FRONT_END_HANDLER = {
  'email was not unique!': 'Sähköposti on jo käytössä',
  'invalid credentials!': 'Sähköposti tai salasana on virheellinen!',
  'invalid request body!': 'Sähköposti tai salasana on virheellinen',
} as const;

type KnownFrontEndErrorTexts = keyof typeof FRONT_END_HANDLER;
```
Kun määrittelyn tekee näin pitää teksti hakea seuraavasti
```ts
    const responseText = e.response.data.toLowerCase();
    const frontendText =
      FRONT_END_HANDLER[responseText as KnownFrontEndErrorTexts];
    if (frontendText) {
      console.error(frontendText);
      return frontendText;
    } else {
```

Reply:
Wow! :open_mouth: Erittäin siisti tapa toteuttaa tuo, kiitos! Ei käynyt mielessäkään toteuttaa tuolla tavalla, toivottavasti tulevaisuudessa tulee :sweat_smile: